protoc-gen-builder/generic/module.go

```diff
@@ -268,7 +272,7 @@ func (m *reportModule) Execute(targets map[string]pgs.File, pkgs map[string]pgs.
 		if err != nil {
 			return nil
 		}
-		packageName := f.Package().ProtoName().String()
+		packageName := adjustPackageName(adjustProtoFileName(extractProtoFileName(f.Name().Split()[0])), f.File().InputPath().Dir().String())
```

Comment:
It's not clear why you do the split here - could something go wrong with this big inline statement? might be better to break it up